docs/design/2022-03-21-delete-support.md

```diff
@@ -0,0 +1,208 @@
+# TiSpark Design Documents
+
+- Author(s): shiyuhang0
+- Tracking Issue: https://github.com/pingcap/tispark/issues/2235
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Motivation or Background](#motivation-or-background)
+- [Goals](#goals)
+- [API](#api)
+- [Detailed Design](#detailed-design)
+  * [Parse filters to WHERE clause](#parse-filters-to-where-clause)
+  * [Query Data ByPass TiDB](#query-data-bypass-tidb)
+  * [Check data and schema](#check-data-and-schema)
+  * [Extract handle](#extract-handle)
+  * [Encode record & index](#encode-record---index)
+  * [2PC](#2pc)
+- [Compatibility](#compatibility)
+- [Test Design](#test-design)
+  * [Functional Tests](#functional-tests)
+  * [Compatibility Tests](#compatibility-tests)
+  * [Benchmark Tests](#benchmark-tests)
+- [References](#references)
+
+## Introduction
+
+New feature: TiSpark Delete Feature
+
+## Motivation or Background
+
+TiSpark is a connector that provides read and write support. However, it lacks the ability to delete. 
+
+Delete operations are becoming more and more important for many big-data workflows. Right now, we are able to provide delete operations after the support of delete in spark 3.0.
+
+## Goals
+**Must have**
+- Support delete bypass TiDB and provide friendly API with spark SQL
+- Compatible with TiDB 4. x and 5. x 
+
+**Nice to have**
+- Optimize the code in the write feature which can be reused. Because write and delete have similar logic. However, the code in the write is hard to reuse.
```

Comment:
The code for write is hard to use **currently** so we need to optimize, right?

Reply:
yes